### ISSUE TITLE
Implement SHAKIRA_DATA module

### ISSUE THREAD

**Issue by OlivierBBB** [Consensys/linea-arithmetization#624]
There are in this module
- 3 different regimes (KECCAK, SHA2-256 and RIPEMD-160)
- every activation of one of these goes through 3 sub phases
  - DATA: variable number of rows (but at least 1)
  - RESULT: 2 rows always
  - EXTRA: 3 rows explained below
- the EXTRA phase is a little stupid it's just to make sure every activation produces at least 4 + 2 = 6 rows in total. Since DATA can be as short as 1 row and RESULT is alwasy 2 rows we need 3 extra rows
- let me explain the "4 + 2 = 6" comment, it comes from
  - 4 bytes required to justify smallness of ID increments
  - 2 bytes required to show smallness of the final nBytes value
- a lot of the time the EXTRA rows are just wasted space

Also INDEX_MAX of the data phase in only **implicitly defined in the spec**. It must be equal to
- 0 when `0 < SIZE <=16 `
- 1 when `16 < SIZE <= 32`
- 2 when `32 < SIZE <= 48`
- etc ...

In other words the INDEX_MAX of the data phase must be set to `(SIZE - 1) / 16`. Also this module is only activated when SIZE is nonzero.
